### DISCUSSION TITLE
Manual Tally of Recent Sourcehut Migration Vote

### DISCUSSION THREAD

**Discussion by emabrey** [tenacityteam/tenacity-legacy#216] [Announcements]
Even though Semisol has already tallied the votes and used some kind of software to do the tally, I wanted to address some concerns people had about using potentially proprietary software and do it by hand here:

### Raw Tally:
```
2,4
2
3
2,1,4
2,4
2,5,1
4
4,2
3
5
1,2
1,5
4
4
5
1,2
2,4
2
2,4,5
3,4
1,5
2
4,3
3
5,4
4
5
5
4,2
5
3
```
### First Pass Count
1: 4
2: 9
3: 5
4: 7
5: 4

### Second Pass Count
We have a tie between 1 and 5, so we have to pick one to remove and the rules aren't clear about how that is handled.
If we remove 5 first:
```
remove 5, remove 1, remove 3: **2 wins**
2: 11
4: 9
```

Alternatively, if we remove 1 first, we've gotten a tie between 3 and 5.
So, repeating that same "taking both choices" approach, let's try removing 1 and comparing the result to removing 5 first.
```
remove 1 -> tie between 3 and 5
2: 11
3: 5
4: 8
5: 5
```

### Third Pass Count
Now we need to look at both options for the 3 and 5 tie within the break of the 1 tie:
```
remove 1 -> remove 3 -> remove 5: 2 wins
2: 11
4: 10
```
```
remove 1 -> remove 5 -> remove 3: 2 wins

2: 11
4: 10
```

### Outcome

So we have 3 possible outcomes for how the rules could play out:

5➡️ 1➡️ 3 yields a 55% win for option 2.
1➡️ 3➡️ 5 yields a 52% win for option 2.
1➡️ 5➡️ 3 yields a 52% win for option 2.

In all three cases option 2 won with one vote over 50%, the minimum possible winning vote. My conclusion is therefore that the community is very evenly split on wanting to stay on GitHub vs wanting to use Sourcehut, even though the actual first-pass-the-post interpretation is that option 2 won (which it technically did, of course).

**Comment by Be-ing** [comment]
What is the point of this vote? Moving away from GitHub without working Windows and macOS builds on anything else would be suicidal for this project.

Furthermore the voting was presented as if it was a binary choice between GitHub and SourceHut but that's not the case. There are several more alternatives that were not presented as options to vote on and only came up late in the discussion after many votes were posted.

**Reply by Songtech-0912** [reply]
The point is that we are choosing which path we want the project to take based on community consensus. Again, this is NOT going to be the deciding factor for what we do. It's supposed instead to find out community opinion as well as to be a forum to share ideas. Voting is **never perfect**, but it's a good way to keep all the voices in a community heard, which is itself essential to keep a community together.

**Reply by fossdd** [reply]
@Be-ing Of course the team tries to keep Windows and macOS builds, maybe per external GitHub CI or try to build it using Linux. (@falkTX plans that i guess)

**Reply by falkTX** [reply]
I already had audacity built with mingw before, tested it and run just fine in both Wine and Windows VM.
Most of the needed changes at https://github.com/audacity/audacity/pull/881 which won't be merged now due to CLA.
still need to get used to sr.ht, but looking around seems pretty simple to setup linux and windows builds.
on macOS we either remain with github actions, or find something else (like https://github.com/sickcodes/Docker-OSX)

**Reply by Be-ing** [reply]
>  on macOS we either remain with github actions, or find something else (like https://github.com/sickcodes/Docker-OSX)

This remains hypothetical until someone does it.

**Reply by Be-ing** [reply]
> community consensus

This does not feel like community consensus to me. This feels like a small minority trying to push SourceHut without presenting the community with the full range of options.

**Reply by unbeatable-101** [reply]
Agreed, by making the vote GitHub, SourceHut, or a Write-in unless people look through the whole thread and see why other options are presented, they will just choose one of the original two. Looking through the thread I saw GitLab, Codeberg, and a self-hosted Gitea as possible alternatives. It is my opinion that there should be a revote with some of the alternative ones listed.

**Comment by Semisol** [comment]
"I wanted to address some concerns people had about using potentially proprietary software" I used the GPLv2 version, no issues on that, fyi.
